lib/src/components/buttons/button_group.dart

```diff
@@ -0,0 +1,254 @@
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: getButtons(),
+    );
+  }
+
+  /// Returns [GroupButton]s with there appropriate styling.
+  List<GroupButton> getButtons() {
+    for (final element in buttons) {
+      element._isInitial = element._isFinal = false;
+    }
+
+    buttons.first._isInitial = true;
+    buttons.last._isFinal = true;
+
+    return buttons;
+  }
+}
+
+/// Group Button item
+// ignore: must_be_immutable
+class GroupButton extends StatefulWidget {
+  /// Constructs [GroupButton]
+  GroupButton({
+    super.key,
+    required this.isLarge,
+    required this.rounded,
+    this.label,
+    this.icon,
+    this.onPress,
+  });
+
+  /// Constructs dropdown group button
+  GroupButton.dropdown(
+      {super.key,
+      required this.isLarge,
+      required this.rounded,
+      required this.onPress,
+      this.icon,
+      this.label,});
+
+  ///Constructs group button with icon
+  GroupButton.icon(
+      {super.key,
+      required this.isLarge,
+      required this.rounded,
+      required this.icon,
+      this.onPress,
+      this.label,});
+
+  /// Determines size of [GroupButton]
+  final bool isLarge;
+
+  /// Determinses border radius of [GroupButton]
+  final bool rounded;
+
+  /// Label for [GroupButton]
+  final String? label;
+
+  /// Optional icon for [GroupButton]
+  final IconData? icon;
+
+  /// Function for when [GroupButton] is clicked.
+  final VoidCallback? onPress;
+
+  /// If [GroupButton] is the first button in its list.
+  bool _isInitial = false;
+
+  /// If [GroupButton] is the final button in its list.
+  bool _isFinal = false;
+
+  @override
+  State<GroupButton> createState() => _GroupButtonState();
+
+  /// Returns copy of [GroupButton] with fields.
+  GroupButton copyWith({bool? isFinal, bool? isInitial}) {
+    return GroupButton(
+      key: key,
+      isLarge: isLarge,
+      rounded: rounded,
+      label: label,
+      icon: icon,
+      onPress: onPress,
+      // isFinal: isFinal ?? this.isFinal,
+      // isInitial: isInitial ?? this.isInitial,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      // ..add(DiagnosticsProperty<bool?>('isFinal', isFinal))
+      ..add(DiagnosticsProperty<bool>('isLarge', isLarge))
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(StringProperty('Label', label))
+      ..add(DiagnosticsProperty<IconData?>('icon', icon))
+      ..add(ObjectFlagProperty<VoidCallback?>.has('onPress', onPress));
+    // ..add(DiagnosticsProperty<bool?>('isInitial', isInitial));
+  }
+}
+
+class _GroupButtonState extends State<GroupButton> {
+  late bool selected;
+  late MaterialStatesController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    selected = false;
+    controller = MaterialStatesController();
+  }
+
+  void onPress() {
+    widget.onPress!();
```

Comment:
This call is not safe, as `!` removes null-safety. To make this call safe we can change it to:
``` widget.onPress?.call();```
as using `?` maintains null-safety.